### DISCUSSION TITLE
Supported animations with Scroll

### DISCUSSION THREAD

**Discussion by redoxahmii** [folke/snacks.nvim#237] [Q&A]
What are the supported animations with scroll?
The default is `linear` and the documentation suggests that there are 45 easing functions supported so i assume the names provided in [this](https://github.com/kikito/tween.lua?tab=readme-ov-file#easing-functions) link in the documentation were supported?

This is what i am passing right now to change:

```lua
scroll = {
        animate = {
          duration = { step = 5, total = 100 },
          easing = "bounce",
        },
      }
```

Which just errors out so any guidance on that aspect is appreciated.

I might be mistaking that the easing functios have to be provided values for duration and velocity so that should be clarified by this as well.

**Comment by redoxahmii** [comment]
I was missing the `in` and `out` before most of the names my bad lol.

I really wish there was auto completion or some intellisense from LSP when using `opts` to pass values.

**Comment by folke** [comment]
![image](https://github.com/user-attachments/assets/34cb97af-5dbe-4471-91f8-fcd0a0048f15)

Be aware though that the bounce and elastic family might cause issues with scroll.

I personally like `inQuad` or `inExpo`, both fast initially and slower at the end

Check the animimate docs for a link to a doc with all the easing functions. (second link)

**Reply by redoxahmii** [reply]
Yes `inQuad` and `inExpo` seem to be the only one that work best.
Thank you.
Might be out of scope of my question but how do you handle intellisense when writing in an `opts` table or you just use the documentation provided by that plugin?

Most plugins don't tend to have proper types so this usually stumps a lot of time when configuring things having to look at the documentation so just wanted to ask.
Thank you again.

**Reply by redoxahmii** [reply]
I would go the `config` route but that does not merge the tables as far as i remember and lazy.nvim also explicitly mentions using `opts` to provide options.

**Reply by folke** [reply]
See my screenshot? That `---@type` line will give you completion.
